gitlab-ls.py

```diff
@@ -130,6 +140,8 @@ def get_timestamp() -> str:
 
     def fetch_projects(self, project_paths: List, progress: WorkProgress) -> None:
         self.report_progress(progress, "Fetching missing projects")
+        if self.client is None:
```

Comment:
This is a nice safeguard to have, but are there cases where we could call that that function before init_gitlab ?

Reply:
I did it order to cleanup to the warnings/errors reported by `pyright`